File: intersection-of-two-arrays/method2/Solution.go
func intersection(nums1 []int, nums2 []int) []int {
	sort.Ints(nums1)
	sort.Ints(nums2)
	var res []int
	p1, p2 := 0, 0
	for p1 < len(nums1) && p2 < len(nums2) {
		x, y := nums1[p1], nums2[p2]
		if x == y {
			if res == nil || x != res[len(res)-1] {
				res = append(res, x)
			}
			p1++
			p2++
		} else if x < y {
			p1++
		} else {
			p2++
		}
	}
	return res
}

File: intersection-of-two-arrays/method2/Solution.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        // 排序 + 双指针思路
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int p1 = 0;
        int p2 = 0;
        int i = 0;

        int[] intersection = new int[nums1.length + nums2.length];
        while(p1 < nums1.length && p2 < nums2.length){
            if(nums1[p1] == nums2[p2]){
                if(i == 0 || intersection[i - 1] != nums1[p1]){
                    intersection[i++] = nums1[p1];
                }
                p1++;
                p2++;
            }else if(nums1[p1] < nums2[p2]){
                p1++;
            }else{
                p2++;
            }
        }
        return Arrays.copyOfRange(intersection,0,i);
    }
}